src/Layouts/SearchLayout.tsx

```diff
@@ -0,0 +1,116 @@
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import SearchBar from '../Components/SearchBar/SearchBar';
+import getProducts, { getProduct } from '../model/apiRoot';
+import CardsContainer from '../Components/CardsContainer/CardsContainer';
+import Spinner from '../Elements/Spinner/Spinner';
+import Paginator from '../Components/Paginator/Paginator';
+import ProductCard from '../Components/ProductCard/ProductCard';
+import Selector from '../Components/Selector/Selector';
+
+function SearchLayout() {
+  const [items, setItems] = useState([]);
+  const [curItem, setCurItem] = useState(null);
+  const [pagesCount, setPagesCount] = useState(1);
+  const [isLoading, setisLoading] = useState(false);
+  const [searchString, setSearchString] = useState('');
+  const [itemsPerPage, setItemsPerPage] = useState(5);
+  const [curPage, setCurPage] = useState(1);
+  const [error, setError] = useState(false);
+
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  useEffect(() => {
+    setisLoading(true);
+    setTimeout(() => {
+      getProducts({
+        search: searchString,
+        limit: itemsPerPage,
+        pageNumber: curPage - 1,
+      }).then((res) => {
+        setPagesCount(Math.ceil(res.total / itemsPerPage) || 1);
+        setItems(res.products);
+        setisLoading(false);
+      });
+    }, 200);
+  }, [searchString, itemsPerPage, curPage, pagesCount]);
+
+  useEffect(() => {
+    setCurPage(1);
+  }, [searchString, itemsPerPage, pagesCount]);
+
+  useEffect(() => {
+    const product = searchParams.get('product');
+    const queryString = new URLSearchParams();
+    if (searchString) queryString.append('search', searchString);
+    if (curPage > 1) queryString.append('page', curPage.toString());
+    if (product) queryString.append('product', product);
+    setSearchParams(queryString);
+  }, [searchString, curItem, curPage, setSearchParams, searchParams]);
+
+  useEffect(() => {
+    const globId = searchParams.get('product') || '';
+    if (globId) {
+      setisLoading(true);
+      setTimeout(() => {
+        getProduct(globId).then((res) => {
+          if (res.id) setCurItem(res || null);
+        });
+        setisLoading(false);
+      }, 200);
```

Comment:
If you wanted to show the loading, then perhaps you should have arranged the show/hide bootloader more correctly
```ts
  setisLoading(true);
  getProduct(globId).then((res) => {
    if (res.id) setCurItem(res || null);
    setisLoading(false);
  });
```